cmd/soroban-rpc/internal/ledgerentry_storage/ledgerentry_storage.go

```diff
@@ -0,0 +1,238 @@
+package ledgerentry_storage
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/stellar/go/historyarchive"
+	"github.com/stellar/go/ingest"
+	backends "github.com/stellar/go/ingest/ledgerbackend"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/go/xdr"
+)
+
+// TODO: Make this configurable?
```

Comment:
yes - good call. no need to expose that to the UI for now, but having this a configurable parameter makes sense.